File: Python/Math-Array-String/74.py
# 2215. Find the Difference of Two Arrays

from typing import List

class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        set1, set2 = set(nums1), set(nums2)
        return [list(set1 - set2), list(set2 - set1)]
    

sol = Solution()
output = sol.findDifference([1,2,3], [2,4,6])
print(output) # Output: [[1, 3], [4, 6]]

File: JavaScript/Math-Array-String/74.js
// 2215. Find the Difference of Two Arrays

var findDifference = function(nums1, nums2) {
    let set1 = new Set(nums1);
    let set2 = new Set(nums2);
    
    let diff1 = [...set1].filter(num => !set2.has(num));
    let diff2 = [...set2].filter(num => !set1.has(num));
    
    return [diff1, diff2];
};

// Example test cases
console.log(findDifference([1,2,3], [2,4,6])); // Output: [[1,3],[4,6]]
console.log(findDifference([1,2,3,3], [1,1,2,2])); // Output: [[3],[]]
